### PR TITLE
Refactor nectar swap execution

### PR DESCRIPTION
**Pull request by thomaseizinger**
This patch serious refactors nectar's swap execution to more generally deal with scenarios in which we want to refund.

We also add logging so we can see what is going on.

Lastly, we refactor `SwapExecutor#execute` to no longer be an async function to avoid accidential blocking of the event-loop.

FYI: @luckysori

### CODE REVIEW

File: nectar/src/main.rs [hunk 1]
```diff
@@ -13,7 +13,7 @@
 #![cfg_attr(not(test), warn(clippy::unwrap_used))]
 #![forbid(unsafe_code)]
 #![recursion_limit = "512"]
-#![type_length_limit = "1944624"]
+#![type_length_limit = "7743004"]
```
Comment (D4nte) [inline nectar/src/main.rs:16]: 💥

File: nectar/src/swap/comit.rs [hunk 2]
```diff
@@ -8,3 +8,18 @@ mod herc20_hbit;
 pub use comit::{ethereum, *};
 pub use hbit_herc20::{hbit_herc20_alice, hbit_herc20_bob};
 pub use herc20_hbit::{herc20_hbit_alice, herc20_hbit_bob};
+
+use std::fmt::Debug;
```
Comment (D4nte) [inline nectar/src/swap/comit.rs:12]: Nectar is in `comit-rs`. If that "could be upstreamed" why not upstreaming it directly? Feels very limbo-ish.

**Review by D4nte — APPROVED** [review]

File: nectar/src/swap/comit.rs (hunk 2 above)
Comment (thomaseizinger) [reply nectar/src/swap/comit.rs:12]: Yeah, I am planning to do that at some point, just didn't want to add even more noise to these PRs :D

File: nectar/src/main.rs (hunk 1 above)
Comment (thomaseizinger) [reply nectar/src/main.rs:16]: Yeah, don't know what is going on here :(

**Pull request merged**